medicines/doc-index-updater/src/models.rs

```diff
@@ -158,7 +158,38 @@ pub struct CreateMessage {
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
 pub struct DeleteMessage {
     pub job_id: Uuid,
-    pub document_content_id: String,
+    #[serde(
+        alias = "document_content_id",
+        deserialize_with = "string_or_unique_document_identifier"
+    )]
+    pub document_id: UniqueDocumentIdentifier,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub enum UniqueDocumentIdentifier {
+    ContentId(String),
+    MetadataStorageName(String),
+}
+
+/// Exists for historic backwards compatibility
+/// Previously, if a DeleteMessage's document_id was set, it was a string, and identified a content_id.
+/// We convert strings to content ids to allow deserialisation of old messages to work.
+/// Also allows for document_manager endpoints to continue to accept `String` from the path, and easily convert these.
```

Comment:
this last line is more relevant to the `From<String>` implementation below than this deserializer

Reply:
line moved